TS: global mutate() is not generic

# Bug report

## Description / Observed Behavior

I cannot set a generic type to `mutate()` calls.

## Expected Behavior

I should be able to set a generic like so:

```ts
mutate<MyType>(...)
```

## Repro Steps / Code Example

```ts
import { mutate } from 'swr';

interface MyType {
  text: string;
}

mutate<MyType>('');
```
```
src/test.ts7:8 - error TS2558: Expected 0 type arguments, but got 1.

7 mutate<MyType>('');
         ~~~~~~
```

## Additional Context

SWR version: 0.5.6
I had tried fixing it in #1069 but it looks like things have changed since then. 😔 

![image](https://user-images.githubusercontent.com/567105/117438683-c8a97200-af29-11eb-9a5e-dc986a20dc50.png)

It returns `Promise<any>` which isn't quite safe.

--- a/src/types.ts
+++ b/src/types.ts
@@ -90,11 +90,20 @@ export type Mutator<Data = any> = (
   shouldRevalidate?: boolean
 ) => Promise<Data | undefined>
 
-export type ScopedMutator<Data> = (
-  key: Key,
-  data?: Data | Promise<Data> | MutatorCallback<Data>,
-  shouldRevalidate?: boolean
-) => Promise<Data | undefined>
+export interface ScopedMutator<Data = any> {
+  /** This is used for bound mutator */
+  (
+    key: Key,
+    data?: Data | Promise<Data> | MutatorCallback<Data>,
+    shouldRevalidate?: boolean
+  ): Promise<Data | undefined>
+  /** This is used for global mutator */
+  <T = any>(
+    key: Key,
+    data?: T | Promise<T> | MutatorCallback<T>,
+    shouldRevalidate?: boolean
+  ): Promise<T | undefined>
+}
 
 export type KeyedMutator<Data> = (
   data?: Data | Promise<Data> | MutatorCallback<Data>,

--- a/src/use-swr.ts
+++ b/src/use-swr.ts
@@ -700,7 +700,8 @@ Object.defineProperty(SWRConfig, 'default', {
 export const mutate = internalMutate.bind(
   null,
   defaultConfig.cache
-) as ScopedMutator<any>
+) as ScopedMutator
+
 export function createCache<Data>(
   provider: Cache
 ): {
@@ -713,5 +714,4 @@ export function createCache<Data>(
     mutate: internalMutate.bind(null, cache) as ScopedMutator<Data>
   }
 }
-
 export default useSWR

